### PR TITLE
Add content-security policy to console,account,oauth

### LINKED ISSUE #4212: Add Content-Security-Policy header to HTML pages
#### Summary

I think we should add a `Content-Security-Policy` header to the HTML pages of the Account app and the Console.

#### How do you propose to implement this?

- `default-src 'self' assets.cloud.thethings.industries [...];` (constructed from `AssetsBaseURL` and `BrandingBaseURL`)
- `connect-src 'self' eu1.cloud.thethings.industries [...] *.ingest.sentry.io;` (constructed from the API `BaseURL`s)
- `base-uri 'self';`
- `frame-ancestors 'none';`

We shouldn't need `unsafe-inline` and `unsafe-eval`, so if it doesn't already work without those, we should probably try and make it work.

We also need to allow outgoing calls

#### How do you propose to test this?

It should be possible to test this locally if you configure `AssetsBaseURL` and `BrandingBaseURL`.

### PR DESCRIPTION
**Pull request by pgalic96**
#### Summary

Closes #4212 

#### Changes

- Add middleware that adds `Content-Security-Policy` header
- Generate and pass nonce to the application

#### Testing

Manual

##### Regressions

If CSP is set too strict, our scripts and assets won't load which will break UI or functionality.

#### Notes for Reviewers

As discussed with some of you, in the bundled console/account file there are many expressions `Function("return this")` which need `unsafe-eval` policy in `default-src` or `script-src` to be set to work. The `unsafe-eval` is not recommended but I don't think there are immediate security concerns regarding this. However we should figure out why are these expressions necessary and how can we replace them. I will file an issue for this.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/account/server.go [hunk 1]
```diff
@@ -66,6 +67,25 @@ func NewServer(c Component, store Store, config oauth.Config) Server {
 		config:  config,
 		store:   store,
 		session: sess.Session{Store: store},
+		csp: map[string][]string{
+			"default-src": {
+				"'self'",
+				config.UI.AssetsBaseURL,
+				config.UI.BrandingBaseURL,
```
Comment (htdvisser) [inline pkg/account/server.go]: Should we de-duplicate these if they're the same, and omit them if empty?

File: pkg/console/console.go [hunk 2]
```diff
@@ -48,6 +49,32 @@ func New(c *component.Component, config Config) (*Console, error) {
 		Component: c,
 		oc:        oc,
 		config:    config,
+		csp: map[string][]string{
+			"default-src": {
+				"'self'",
+				config.UI.AssetsBaseURL,
+				config.UI.BrandingBaseURL,
```
Comment (htdvisser) [inline pkg/console/console.go:56]: Same. Should we de-duplicate these if they're the same, and omit them if empty?

File: pkg/console/console.go [hunk 3]
```diff
@@ -48,6 +49,32 @@ func New(c *component.Component, config Config) (*Console, error) {
...
+			"connect-src": {
+				"'self'",
+				config.UI.StackConfig.GS.BaseURL,
+				config.UI.StackConfig.IS.BaseURL,
+				config.UI.StackConfig.JS.BaseURL,
+				config.UI.StackConfig.NS.BaseURL,
+				config.UI.StackConfig.AS.BaseURL,
+				config.UI.StackConfig.EDTC.BaseURL,
+				config.UI.StackConfig.QRG.BaseURL,
+				config.UI.StackConfig.GCS.BaseURL,
```
Comment (htdvisser) [inline pkg/console/console.go:68]: Same

File: pkg/webui/csp.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"strings"
+)
+
+// GenNonce returns a nonce used for inline scripts
+func GenerateNonce() string {
```
Comment (htdvisser) [inline pkg/webui/csp.go]: Godoc "GenNonce" does not match func name "GenerateNonce"

File: pkg/webui/csp.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	var b [20]byte
+	if _, err := rand.Read(b[:]); err != nil {
+		panic(err)
+	}
+	return base64.StdEncoding.EncodeToString(b[:])
+}
+
+func GenerateCSPString(csp map[string][]string, nonce string) string {
```
Comment (htdvisser) [inline pkg/webui/csp.go:50]: Missing Godoc

**Comment by pgalic96** [comment]
We need a CSP for `oauth` app as well since it is using the same template as `console` and `account`. Since `account` and `oauth` share the config they can share the csp as well.

**Review by htdvisser — APPROVED** [review]

**Review by kschiffer — APPROVED** [review]
LGTM. Please test this thoroughly once deployed on staging.

**Pull request merged**